modules/ting/src/Search/TingSearchResultInterface.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * @file
+ * The TingSearchResultInterface interface.
+ */
+
+namespace Ting\Search;
+
+/**
+ * Interface TingSearchResultInterface
+ *
+ * @package Ting\Search
+ */
+interface TingSearchResultInterface {
```

Comment:
I donno about this.

`getNumTotalCollections()` and `getNumTotalObjects()` are to closely named as to what they really are in opensearch. `numTotalObjects` is the number of object hits in the search, while `numTotalCollections` is the number of collections in the reply (mostly capped by the step value (search count)).

`count()` (number of collections in result) and `hits()` (total hits) would make sense. Though, `numTotalCollections` is only used in two places, the pager https://github.com/ding2/ding2/blob/master/modules/ting_search/plugins/content_types/ting_search.inc and https://github.com/ding2/ding2/blob/master/modules/p2/ding_message/ding_message.module#L300 .

The pager should probably use the search page size instead (which isn't available in the opensearch result, but as we're moving into objects here, the provider could just copy it from the query to the result). 

`ding_message` just wants the number of collections from the result, and I think everybody else is using count(TingClientSearchResult::collections).

Not too fond of `getTingEntityCollections()` either, `getCollections()` are fine, or have you considered ArrayAccess and Countable interfaces?